packages/core/test/utils.test.ts

```diff
@@ -79,3 +92,139 @@ describe('Utils - getRoutes', () => {
     await expect(getRoutes(111, Environment.DEVNET)).rejects.toThrow('Invalid environment');
   });
 });
+
+describe('Address Validation Utils', () => {
+  describe('isValidSubstrateAddress', () => {
+    it('should validate a correct Substrate address', () => {
+      const validAddress = '1v72SmRT7XHwSa6hpEhUrgspM9bpmDRNw3F4wucb2GwkynQ';
+      expect(isValidSubstrateAddress(validAddress)).toBe(true);
+    });
+
+    it('should return false for an invalid Substrate address', () => {
+      const invalidAddress = 'invalidAddress';
+      expect(isValidSubstrateAddress(invalidAddress)).toBe(false);
+    });
+  });
+
+  describe('isValidEvmAddress', () => {
+    it('should validate a correct EVM address', () => {
+      const validAddress = '0x742d35Cc6634C0532925a3b844Bc454e4438f44e';
+      expect(isValidEvmAddress(validAddress)).toBe(true);
+    });
+
+    it('should return false for an invalid EVM address', () => {
+      const invalidAddress = 'invalidAddress';
+      expect(isValidEvmAddress(invalidAddress)).toBe(false);
+    });
+  });
+
+  describe('isValidBitcoinAddress', () => {
+    describe('For Bitcoin mainnet', () => {
+      beforeEach(() => {
+        process.env.SYGMA_ENV = Environment.MAINNET;
+      });
+
+      it('should validate a correct P2TR Bitcoin address', () => {
+        const p2trAddress = 'bc1p5d7rjq7g6rdk2yhzks9smlaqtedr4dekq08ge8ztwac72sfr9rusxg3297';
+        expect(isValidBitcoinAddress(p2trAddress)).toBe(true);
+      });
+
+      it('should validate a correct P2WPKH Bitcoin address', () => {
+        const p2wpkh = 'bc1qar0srrr7xfkvy5l643lydnw9re59gtzzwf5mdq';
+        expect(isValidBitcoinAddress(p2wpkh)).toBe(true);
+      });
+
+      it('should validate a correct P2SH Bitcoin address', () => {
```

Comment:
AFAIK and you can confirm this with @MakMuftic we just need to test P2TR and P2WPKH. Same for testnet